### ISSUE TITLE
service-binding-operator - crunchyDB operator populating tls.key with wrong permissions 

### ISSUE THREAD

**Issue by pacostas** [redhat-developer/service-binding-operator#1157]
This issue is not related 100% to service-binding-operator

### What is the environment (Minikube, Openshift)?
Openshift

### What is the SBO version used?
apiVersion: binding.operators.coreos.com/v1alpha1

### What are the steps to reproduce this issue?
  1.  Deploying [cruncry db](https://access.crunchydata.com/documentation/postgres-operator/5.1.0/tutorial/) in openshift
  2. Deploying a simple nodejs app
  3. Using a service binding operator to bind the app with node js app
  4. Both services are being bound successfully after creating service binding, and on env variables of nodejs app, a SERVICE_BINDING_ROOT env has been set.
  5. under SERVICE_BINDING_ROOT/myapp are all files populated from crunchy db and ready to use, in order to connect with the Database.

### What is the expected behaviour?
Under SERVICE_BINDING_ROOT/myapp directory, having tls.key with proper permissions which in our case is 0600

### What is the actual behaviour?
tls.key has user or world group permissions.

### Postresql logs
Postgresql throws below error:
```
private key file "/bindings/myapp/tls.key" has group or world access; permissions should be u=rw (0600) or less\n
```

**Comment by mhdawson** [comment]
I looked at the service binding standard and I did not see any mention of file permissions. I think this might be a gap as illustrated by this problem where some programs (like OpenSSL) enforce limited access rights on keys. 

The initial question is if the service binding operator/service binding spec provides a way to specify the access rights for a key, if so then possibly the crunchy binding needs to be updated, if not that may be a gap in the spec/the service binding operator implementation.

**Comment by sadlerap** [comment]
AFAIK there's no way to set permissions on projected binding data solely from a `ServiceBinding` resource; neither the `binding.operators.coreos.com` nor the `servicebinding.io` resources expose this functionality.  As a workaround for now, you can override permissions in the volume specification we write (e.g. set `.spec.template.spec.volumes[].secret.defaultMode` in a `deployment` to 0600).

IMO this is a shortcoming of both SBO and its upstream spec.  For SBO, it'd probably make sense to add a field to specify the default permissions for the data we project.  As for the spec, I'd imagine a similar approach would work as well, though it'll probably have to start as an extension first.

**Comment by pacostas** [comment]
> AFAIK there's no way to set permissions on projected binding data solely from a `ServiceBinding` resource; neither the `binding.operators.coreos.com` nor the `servicebinding.io` resources expose this functionality. As a workaround for now, you can override permissions in the volume specification we write (e.g. set `.spec.template.spec.volumes[].secret.defaultMode` in a `deployment` to 0600).
> 
> IMO this is a shortcoming of both SBO and its upstream spec. For SBO, it'd probably make sense to add a field to specify the default permissions for the data we project. As for the spec, I'd imagine a similar approach would work as well, though it'll probably have to start as an extension first.

Thanks for the workaround. I did an attempt to change that on deployment.yaml while the pod was already running and unfortunately ended up with:
```
Error "Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
  core.PodSpec{
   Volumes: []core.Volume{
   {
   Name: "odbc-psql-app-d-hippo-pc",
   VolumeSource: core.VolumeSource{
   ... // 3 identical fields
   AWSElasticBlockStore: nil,
   GitRepo: nil,
   Secret: &core.SecretVolumeSource{
   SecretName: "odbc-psql-app-d-hippo-pc-59924545",
   Items: nil,
-  DefaultMode: &384,
+  DefaultMode: &420,
   Optional: nil,
   },
   NFS: nil,
   ISCSI: nil,
   ... // 21 identical fields
   },
   },
   {Name: "kube-api-access-jm2hw", VolumeSource: {Projected: &{Sources: {{ServiceAccountToken: &{ExpirationSeconds: 3607, Path: "token"}}, {ConfigMap: &{LocalObjectReference: {Name: "kube-root-ca.crt"}, Items: {{Key: "ca.crt", Path: "ca.crt"}}}}, {DownwardAPI: &{Items: {{Path: "namespace", FieldRef: &{APIVersion: "v1", FieldPath: "metadata.namespace"}}}}}, {ConfigMap: &{LocalObjectReference: {Name: "openshift-service-ca.crt"}, Items: {{Key: "service-ca.crt", Path: "service-ca.crt"}}}}}, DefaultMode: &420}}},
   },
   InitContainers: nil,
   Containers: {{Name: "odbc-psql-app", Image: "image-registry.openshift-image-registry.svc:5000/postgres-operat"..., Ports: {{ContainerPort: 8080, Protocol: "TCP"}}, Env: {{Name: "SERVICE_BINDING_ROOT", Value: "/bindings"}}, ...}},
   ... // 28 identical fields
  }
" for field "spec".
``` 
Also I couldnt find somewhere this ```.spec.template.spec.volumes[].secret.defaultMode``` option. If you have more context about this option, it would be a nice to share this solution :)

**Comment by baijum** [comment]
> We are working on a fix here: #1182 This is scheduled for 1.2 release.

Due to the [Kubernetes bug](https://github.com/kubernetes/kubernetes/issues/57923), we cannot set permission from the SBO controller. So, we decided to document a workaround in another PR #1191
